src/Control/Monad/Freer/State.hs

```diff
@@ -94,14 +98,9 @@ transactionState
   -> Eff effs a
 transactionState m = do
     s0 <- get @s
-    (x, s) <- interposeS s0 (\s x -> pure (x, s)) handle m
+    (x, s) <- interceptS stateNat s0 m
```

Comment:
I don't a million percent understand what `transactionState` is supposed to do, but presumably this does the same thing.